controller/src/main/java/org/apache/rocketmq/controller/impl/DefaultBrokerHeartbeatManager.java

```diff
@@ -61,42 +60,37 @@ public void start() {
     @Override
     public void shutdown() {
         this.scheduledService.shutdown();
-        this.executor.shutdown();
     }
 
     public void scanNotActiveBroker() {
         try {
-            log.info("start scanNotActiveBroker");
             final Iterator<Map.Entry<BrokerAddrInfo, BrokerLiveInfo>> iterator = this.brokerLiveTable.entrySet().iterator();
             while (iterator.hasNext()) {
                 final Map.Entry<BrokerAddrInfo, BrokerLiveInfo> next = iterator.next();
-                long last = next.getValue().getLastUpdateTimestamp();
-                long timeoutMillis = next.getValue().getHeartbeatTimeoutMillis();
+                BrokerLiveInfo brokerLiveInfo = next.getValue();
+                long last = brokerLiveInfo.getLastUpdateTimestamp();
+                long timeoutMillis = brokerLiveInfo.getHeartbeatTimeoutMillis();
                 if ((last + timeoutMillis) < System.currentTimeMillis()) {
-                    final Channel channel = next.getValue().getChannel();
+                    final Channel channel = brokerLiveInfo.getChannel();
                     iterator.remove();
                     if (channel != null) {
                         RemotingHelper.closeChannel(channel);
                     }
-                    this.executor.submit(() ->
-                        notifyBrokerInActive(next.getKey().getClusterName(), next.getValue().getBrokerName(), next.getKey().getBrokerAddr(), next.getValue().getBrokerId()));
-                    log.warn("The broker channel {} expired, brokerInfo {}, expired {}ms", next.getValue().getChannel(), next.getKey(), timeoutMillis);
```

Comment:
IMO, it is necessary to notify the election ASAP, rather than relying solely on the timing task.